de.metas.banking/de.metas.banking.base/src/main/java/de/metas/banking/payment/impexp/CPaymentImportTableSqlUpdater.java

```diff
@@ -0,0 +1,117 @@
+package de.metas.banking.payment.impexp;
+
+import org.adempiere.ad.trx.api.ITrx;
+import org.compiere.model.I_I_DiscountSchema;
+import org.compiere.util.DB;
+import org.slf4j.Logger;
+
+import de.metas.logging.LogManager;
+import de.metas.pricing.impexp.DiscountSchemaImportProcess;
+import lombok.NonNull;
+import lombok.experimental.UtilityClass;
+
+/*
+ * #%L
+ * de.metas.adempiere.adempiere.base
+ * %%
+ * Copyright (C) 2017 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+/**
+ * A helper class for {@link DiscountSchemaImportProcess} that performs the "dirty" but efficient SQL updates on the {@link I_I_DiscountSchema} table.
+ * Those updates complements the data from existing metasfresh records and flag those import records that can't yet be imported.
+ *
+ * @author metas-dev <dev@metasfresh.com>
+ *
+ */
+@UtilityClass
+public class CPaymentImportTableSqlUpdater
+{
+	private static final transient Logger logger = LogManager.getLogger(CPaymentImportTableSqlUpdater.class);
+
+	public void updatePaymentImportTable(@NonNull final String whereClause)
+	{
+		dbUpdateBPartners(whereClause);
+		dbUpdateInvoices(whereClause);
+		dbUpdateIsReceipt(whereClause);
+		dbUpdateErrorMessages(whereClause);
+	}
+
+	private void dbUpdateBPartners(@NonNull final String whereClause)
```

Comment:
imho we need to include AD_Org_ID in the whereclauses `C_BPartner.Value`, `C_Invoice.DocumentNo` etc can be ququal among different orgs